src/server/repositories/UrlRepository.ts

```diff
@@ -197,21 +200,20 @@ export class UrlRepository implements UrlRepositoryInterface {
     }
   }
 
-  public getLongUrl: (shortUrl: string) => Promise<string> = async (
-    shortUrl,
-  ) => {
-    try {
-      // Cache lookup
-      return await this.getLongUrlFromCache(shortUrl)
-    } catch {
-      // Cache failed, look in database
-      const longUrl = await this.getLongUrlFromDatabase(shortUrl)
-      this.cacheShortUrl(shortUrl, longUrl).catch((error) =>
-        logger.error(`Unable to cache short URL: ${error}`),
-      )
-      return longUrl
+  public getLongUrl: (shortUrl: string) => Promise<RedirectDestination> =
```

Comment:
we need to relax this return condition to `RedirectDestination | string` due to the previously mentioned issue of data format changing

Reply:
Let me do a split of the PRs so they get released separately.